webapp/lib/app/configurator/tags/view.dart

```diff
@@ -157,7 +157,18 @@ class ConfigureTagView extends TagView {
     draggableTag..onDragStart.listen((_) => dragInProgress = true)..onDragEnd.listen((_) => dragInProgress = false);
 
     onEdit = (text) {
-      _view.appController.command(TagsConfigAction.renameTag, new TagData(tagId, text: text));
+      var requestRenameTagData = new RequestRenameTagData(text, () {
```

Comment:
I'd rather we didn't define these as callbacks on a `*Data` object, it's not really a pattern we've used before, and I'm not sure is very easy to read.

Can I suggest instead to:
- change this method to how it was, except that the config action will remain `requestRenameTag`
- add another method for the `ConfigureTagView` class called `showDuplicationWarning` that contains the `onDuplicate` callback code
- the controller code will then call `showDuplicationWarning()` on this tag if duplicate, otherwise will call the `command()` with the `renameTag` action

What do you think?

Reply:
Thanks Mariana, I took the pattern from the JS land - but it makes sense to do this since its an all new pattern.

I have updated the code, except one minor change from what you have suggested - to trigger the modal from the group view instead of tagView directly. Please let me know if that is okay